src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Initializers/WebJobsTelemetryInitializer.cs

```diff
@@ -19,8 +19,9 @@ internal class WebJobsTelemetryInitializer : ITelemetryInitializer
         private static readonly string _currentProcessId = Process.GetCurrentProcess().Id.ToString();
         private readonly string _sdkVersion;
         private readonly string _roleInstanceName;
+        private readonly ApplicationInsightsLoggerOptions _options;
 
-        public WebJobsTelemetryInitializer(ISdkVersionProvider versionProvider, IRoleInstanceProvider roleInstanceProvider)
+        public WebJobsTelemetryInitializer(ISdkVersionProvider versionProvider, IRoleInstanceProvider roleInstanceProvider, ApplicationInsightsLoggerOptions options)
```

Comment:
This should use `IOptions<ApplicationInsightsLoggerOptions>` so it gets auto-injected. You won't need the factory above then.

Reply:
Good call, fixed